lib/src/stream_extensions.dart

```diff
@@ -54,4 +54,22 @@ extension StreamExtensions<T> on Stream<T> {
     });
     return completer.future;
   }
+
+  /// Eagerly listens to this stream and buffers events until needed.
+  ///
+  /// The returned stream will emit the same events as this stream, starting
+  /// from when this method is called. The events are delayed until the returned
+  /// stream is listened to, at which point all buffered events will be emitted
+  /// in order, and then further events from this stream will be emitted as they
+  /// arrive.
```

Comment:
So, add something like:
```dart
/// 
/// The buffering will retain all events until the returned stream is listened to.
/// Therefore the returned stream *must* be listened to, otherwise the buffer
/// can grow arbitrarily large.
/// If the stream is not needed anyway, it should be discarded by doing, at least,
/// `stream.listen(null).cancel();`.

Reply:
I'm not sure "*must* be listened to" is correct—in a lot of cases, this is going to be consuming output from a process or a file where the size is extrinsically constrained and just calling `.bufferUntilListen()` isn't any riskier than something like `File.readAsString()` or `Process.run()`. I rewrote it a bit to make that clearer.

Reply:
That looks fine.